templates/zerver/app/compose.html

```diff
@@ -22,26 +22,23 @@
                     </button>
                 </span>
                 <span class="new_message_button">
-                    <button type="button" class="button small rounded compose_stream_button"
-                      id="left_bar_compose_stream_button_big"
-                      title="{{ _('New topic') }} (c)">
-                        <span class="compose_stream_button_label">{{ _('New topic') }}</span>
-                    </button>
-                </span>
-                {% if not embedded %}
-                <span class="new_message_button">
-                    <button type="button" class="button small rounded compose_private_button"
-                      id="left_bar_compose_private_button_big"
-                      title="{{ _('New private message') }} (x)">
-                        <span class="compose_private_button_label">{{ _('New private message') }}</span>
-                    </button>
-                </span>
-                {% endif %}
-                <span class="new_message_button">
-                    <button type="button" class="button small rounded compose_reply_button"
+                    <button type="button" class="button small rounded has-dropdown compose_reply_button"
                       id="left_bar_compose_reply_button_big"
                       title="{{ _('Reply') }} (r)">
                         <span class="compose_reply_button_label">{{ _('Reply') }}</span>
+                    </button><!-- we can't have a space character between these divs or they'll have a gap between
+                 --><button type="button" class="button small dropdown compose_dropdown_button">
```

Comment:
This is ugly, but having a space character between the divs would require a "guess and check" negative margin to fix (or having a parent element with font-size: 0), both of which are uglier than this physically ugly but semantically proper approach.

Reply:
You should be able to do whitespace control more nicely using http://jinja.pocoo.org/docs/2.10/templates/#whitespace-control

Reply:
As far as I understand it (which is limited!), jinja whitespace control would apply to whitespace before block elements, but wouldn't affect other whitespace, so it wouldn't be super useful here

Reply:
I think Tim is referring to the following?
![image](https://user-images.githubusercontent.com/890911/57964079-d5b79b80-78e3-11e9-8978-72d0d5722d90.png)

Reply:
(had a comment here, deleted)

Reply:
You can use whitespace control with Jinja comments. (Documentation: “If you add a minus sign (`-`) to the start or end of a block (e.g. a For tag), **a comment**, or a variable expression, the whitespaces before or after that block will be removed”.)

```html
    </button>{# -#}
    <button …>
```